### DISCUSSION TITLE
Optimistic locking using versions

### DISCUSSION THREAD

**Discussion by ryands17** [tywalch/electrodb#295] [Q&A] [answered]
I wanted to know if it's possible to achieve optimistic locking with a version number using ElectroDB. Something similar to what this [doc](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBMapper.OptimisticLocking.html) shows.

**Comment by tywalch** [comment, marked as answer]
Hi @ryands17 👋 

This could be accomplished with the addition of a [version number attribute](https://electrodb.fun/?#code/PQKgBAsg9gJgpgGzARwK5wE4Es4GcA0YuccYGeqCALgUQBYCG5YA7llXWAGbZwB2MXGBDAAUKKwBbAA5QMVMAG8wAUT5V2AT0IBhcgypw1GqpoCShyYWNaz8dVi44MQgL7cMUSWABEiOADGVJ4wAEY+ANziAVB8uApUDKEIpAC8vppQqBgA+onJcDl8DJJwkeKgYAAqDLgA1kI2psJiMXEKVXRYfADmYOl8cCyqDqYAFKJgSpNTYJKwiABc07Oz-Caayz4c3T0++DOrAG6YuFixWwCM+4ezxBhHWAFwW-FyZQerrp+zBsFYoVQhlwy0UtymDACMVQ6jsoPBs1M0hevni2F6N1WU2+COKpXhWMRmmRr3+GJ+hPIaCw5Bgy2C6ApsxxhPguAC2GkGguK0JYCRKJ8aN2mL5VNQNLgdP5GEZCJZWKODAQ6AJfIFywA5MLepqmatxZLpQy4PqwArjqdznw1YSNWBNXxUJJQpg9QipobafTZaaPWB4FwGJQqMsAAxmi2zDlwAxSgCCod5duJgqdLswotZcCDIeWYwAlP0AHxgAAicYAdHwoCxC2apsBgGAAgw+DWFK65rBHDgYGAGFxDBgW-pDDB-foYAB5PgITYy9Dys2oaQwOMwRO2rH2nzp10YLNYpusAwBThcOQtxi9PD8qADviaAdUf6Awz+lhnuhbEBH1YnsQCgBNk5DqPyUh4IkMisHQ-BgKu67jv6QH5kWqSlhWhjVrW9aTrGM5zguJryrcUbdPAAAeeDblM7BwJIILJli0h1LRWJOIg0o+Kx-5YjEMhQGchjLAA2j4kLQrCMA+AAuv6UarPU7GrJxCDcfUfEAc2MZxi2XiyMJpB1HAmhCJeI7SEwGjKkQcgKCZz5oJgOC4P6UwCYZ9FiXuJRlPJfLmqRXwzAqyj5CkQUFlEogCtUXS9HY6y9qc-QjBsSUOJxLgADwClAXDxbsxZRHFeixoYnS7BYDFpeVcZNOYlh5amBVFb0JXiGVY5wFVvTTly1pCOk06SOwOX1ZVCU9DVVgOicLjWpqYAAD4Orp46Jsta2aohG5bZ1oi1JofABNwMJBNao4VXAs1jFAg2xExk29dNA3cnERZgo2zaoMQvgbWU95gPwuDZKQHCkPR3gwFAd4diDlFYPEsHwRtuxgIMwzQ25npwFQ2R8O1PSVoD92PZ9lY9FAhZRK4XWpmAACqa5xgAasq6DvUNaV9TN9gaNlQgAGRKGAMygCAhzgJ0pBKiqpCuhjuAMCcE6zCIMzy6qRBkj0ETi42IBSxr8Vy1asRgErvTdjAKXq1MmseaB6xsxbNqY86B7i-TR24CdZ1cBdH0Iazhh3Q9H1MSzSFwBzCvc09X0zG0KPKJJWTSYQeKmmA2u5yBGBgVQbuLZb7jpJHQ1RD9CH-Tx34+MDoPg-ycFgNDYCZKgA7MHtGg23ASPxLjZD44TxOVlZVDnmM6dQpnVB2NnfnmgWtyAfjbekFjeec3AtyVkBc97wra8b823QxqU4GQ3n7uHwwMAwCfC1nDylzn9pp9YLH2-32XIm0hPCPHgP2ZGmMoAKHiMqA+qxKwsDguQMYr93bmkIMoagpBXDoVLFgsYb9rSEELsXUu78+AFnXt-IegQgSkDbP2cgBMMBEwQHGFGhDYiH2pifcguBZBxBRJqZUCAihDGWjgum0Qep3W+pjPyWpoYAFowxhkuO6CEC8YRL2lMIqEVAVFqI0QGPAHIsAUy1AwEx7JOQhwsv-aGxj85agABIAE0dDTgAJwAHEWAAHYABiYZNSiEkeIfut1LAnxmDnRRlhDHqJ+BnbRcIHSSQMaopJWt95anPHUaQAB1AA0tRAAzFUYxJDXbu3DAcSRQA) or using an [updatedAt attribute](https://electrodb.fun/?#code/PQKgBAsg9gJgpgGzARwK5wE4Es4GcA0YuccYGeqCALgUQBYCG5YA7llXWAGbZwB2MXGBDAAUKKwBbAA5QMVMAG8wAUT5V2AT0IBhcgypw1GqpoCShyYWNaz8dVi44MQgL7cMUSWABEiOADGVJ4wAEY+ANziAVB8uApUDKEIpAC8vppQqBgA+onJcDl8DJJwkeKgYAAqDLgA1kI2psJiMXEKVXRYfADmYOl8cCyqDqYAFKJgSpNTYJKwiABc07Oz-Caayz4c3T0++DOrAG6YuFixWwCM+4ezxBhHWAFwW-FyZQerrp+zBsFYoVQhlwy0UtymDACMVQ6jsoPBs1M0hevni2F6N1WU2+COKpXhWMRmmRr3+GJ+hPIaCw5Bgy2C6ApsxxhPguAC2GkGguK0JYCRKJ8aN2mL5VNQNLgdP5GEZCJZWKODAQ6AJfIFywA5MLepqmatxZLpQy4PqwArVhy4AYpQBBKhqwka3x8VCSUKYUWsuBcBiUB1gMYASn6AD4wAARG0AOj4UBYwbNU2AwDAAQYfDjCg9c1gjhwMDADC4hgwaf0hhgCKm+hgAHk+AhNjL0PKzahpDAbTB7Y6sc6fK73Z6k2AU6wDAFOFw5GnGL08PyoEW+Joi1R-oDDNWJ1Qp1sQF6sePiAoAtlyOp+VI8IkZKw6PwwB2u5Wd6flsGw5GY3GE0HR1rBsm3pWU4HlW4LTAbp4AADzwPspnYOBJBBXlCWkOpEKxJxEGlHxMKPQkYhkKAzkMZYAG0fEhaFYRgHwAF0dyg1Z6mw1ZcIQfD6iI49UytG00y8WRyNIOo4E0IQZzLaQmA0ZUiDkBQJLXNBMBwXAdymEjROQqjBxKMpmL5c0IK+GYFWUfIUjMoMolEAVqi6Xo7HWfNTn6EYNjchxcJcAAeAUoC4ZzdlDKInL0a1DE6XYLBQrzoptJpzEsILiTgEKwt6CLxCiis4Di3o6y5c44i8utJHYALktilyegSqwwE1QTK3tTUwAAHxal9uw6vLRFqTQ+ACbgYSCcryxiuAmrGKAytiNC6qKhrSu5OIQzBZNU1QYhfDasolzAfhcGyUgOFIZDvBgKBFyzE7YKweIHyfNrdjAQZhmurSazgKhsj4HKemjQ75sWzbox6KBgyiVx8sysAAFVOxtAA1ZV0HW8qhHSYrGvsDR-KEAAyJQwBmUAQEOcBOlIJUVVID0PtwBgTirWYRBmBnVSIMkegiCnkxAanOec0gEFqBQoFCe52efVH2oUHnSBk-lH2gywwGZ3pcxgDyOamLmpkl+IpRR187QDIcPTLUR4aG3ARrGrgJo2hXLbmhaNrQi30cxuBsaWraZjaF7lForJ6MIPFTTAFXCFNys-aV80vO9nGoh2599oIyc6B8Y7TvO9Wrq1zJUCLZg+o0XW4Ce+JfrIf7AeB6M5L3OgxgjqEo6oOwY6M80g1uE9-tLz6hnjgPbmjU9u+nxnh9H1MGawS2J5OFwpukTxHngQtns+qAFHiZVwNWaMWEfcgxgXmurfNQhlGoUhXBDVJw1fsYH57KhE6lubRWVsgwj1WOOeugQgSkAzIWcgAMMBA0lsCZWpxyqz2hgvcguBZBxBRJqZUCAihDE6u-OG0RCpzW2p9IyWproAFoAAMjDLh6hmJHGE-dpQEKhFQJhLC2FTDZByLAEMtQMDAMIzk7s1aXU1ihQRi9eaagABIAE0dB1gAJwAHEWAAHYABijDNT23suIB+VChY0PxC1BhzDWE-A4fRcRvD+GOO5gHLUU46jSAAOoAGl4IAGYqiKKTkAy2f9liXAAGxaIAKyXASYw-R+iABMcSEkHDIUAA) 👍 

Let me know if this helps!

**Reply by ryands17** [reply]
Works for me! Thanks a lot 🙌
